### PR TITLE
Implementation of enhancement #17

### LINKED ISSUE #17: issue/0017: Calculating the average, min and max for Durations
## Enhancement

This issue proposes the introduction of new methods for computing the average, min, and max Duration from a collection of Duration:

Example 1:
```` java
Duration duration1 = Duration.of(500, TimeUnit.MILLISECONDS);
Duration duration2 = Duration.of(1, TimeUnit.SECONDS);
List<Duration> durations = Arrays.asList(duration1, duration2);

Duration average = DurationUtils.average(durations);
// result: 0.75 second(s)
````

Example 2:
```` java
Duration min = DurationUtils.min(durations);
// result: 0.5 second(s)
````

Example 3:
```` java
Duration max = DurationUtils.max(durations);
// result: 1.0 second(s)
````

## Additional details
- The code shall be developed in the `issue/0017` branch.

## Quality criteria
The following criteria will be observed during the Code Review:

#### Code coverage
- The produced code shall be secured by unit tests.
- Test coverage shall remain the same or increase.

#### Documentation
- All methods shall be documented with **Javadoc**, providing examples of usage.
- Javadoc pages shall be compilable with no errors or warnings
- README.md shall be updated if applicable

### PR DESCRIPTION
**Pull request by oswaldobapvicjr**

### CODE REVIEW

File: src/test/java/net/obvj/performetrics/util/DurationUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.junit.Test;
+
+public class DurationUtilsTest
+{
```
Comment (oswaldobapvicjr) [inline src/test/java/net/obvj/performetrics/util/DurationUtilsTest.java:19]: Add JUnit test to secure that this class cannot be instantiated.

File: src/test/java/net/obvj/performetrics/util/DurationUtilsTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    public void average_twoOrMoreElements_meanValue()
+    {
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_1_SECOND)), is(equalTo(D_750_MILLIS)));
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_2_SECONDS)), is(equalTo(D_1250_MILLIS)));
+        assertThat(average(Arrays.asList(D_500_MILLIS, D_500_MILLIS, D_2_SECONDS)), is(equalTo(D_1_SECOND)));
+    }
+
+    @Test
+    public void average_containingNullElements_nullElementsDontCount()
+    {
```
Comment (oswaldobapvicjr) [inline src/test/java/net/obvj/performetrics/util/DurationUtilsTest.java:61]: Add a test to secure the behavior when the collection contains only null elements
